This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Boris Nagaev <bnagaev@gmail.com>
Date: Tue, 31 Jan 2017 02:03:55 +0100
Subject: [PATCH 1/1] fix EPOXY_IMPORTEXPORT for static targets

Based on https://github.com/mxe/mxe/issues/1647#issuecomment-275967915

diff --git a/include/epoxy/gl.h b/include/epoxy/gl.h
index 1111111..2222222 100644
--- a/include/epoxy/gl.h
+++ b/include/epoxy/gl.h
@@ -71,8 +71,19 @@
 #define GLAPI extern
 #endif

-#define KHRONOS_APIENTRY __stdcall
-#define KHRONOS_APICALL __declspec(dllimport) __stdcall
+#  ifdef EPOXY_SHARED
+#    define KHRONOS_APIENTRY __stdcall
+#    ifdef EPOXY_DLL
+#      define KHRONOS_APICALL __declspec(dllexport)
+#    else
+#      define KHRONOS_APICALL __declspec(dllimport)
+#    endif
+#  elif EPOXY_STATIC
+#    define KHRONOS_APIENTRY
+#    define KHRONOS_APICALL
+#  else
+#    error "Please define EPOXY_STATIC or EPOXY_SHARED"
+#  endif

 #endif /* _WIN32 */
